app/services/charge_param_retriever.js

```diff
@@ -7,11 +7,14 @@ const cookie = require('../utils/cookies')
 
 exports.retrieve = req => {
   const chargeId = req.params.chargeId ? req.params.chargeId : req.body.chargeId
-  setLoggingField(req, PAYMENT_EXTERNAL_ID, chargeId)
 
+  setLoggingField(req, PAYMENT_EXTERNAL_ID, chargeId)
+  
   if (!chargeId) {
     logger.error('ChargeId was not found in request', getLoggingFields(req))
     return false
+  } else if (req.originalUrl && req.method.toLowerCase() === 'post' && req.originalUrl.includes(`/card_details/${chargeId}/3ds_required_in`)) {
```

Comment:
probably safer to assert that the `frontend_state` and `frontend_state_2` cookies are not present too.

Reply:
done